File: java/json-transform/src/test/java/co/nlighten/jsontransform/functions/TransformerFunctionJwtParseTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.BaseTest;
import org.junit.jupiter.api.Test;

public class TransformerFunctionJwtParseTest extends BaseTest {

    @Test
    void inline() {
        var testJWT = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c";
        assertTransformation(testJWT, "$$jwtparse:$", fromJson("""
{
  "sub": "1234567890",
  "name": "John Doe",
  "iat": 1516239022
}"""));
        //invalid
        assertTransformation(true, "$$jwtparse:$", null);
    }

    @Test
    void object() {
        var testJWT = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c";
        assertTransformation(testJWT, fromJson("""
{
  "$$jwtparse": "$"
}"""), fromJson("""
{
  "sub": "1234567890",
  "name": "John Doe",
  "iat": 1516239022
}"""));
        //invalid
        assertTransformation(true, fromJson("""
{
  "$$jwtparse": "$"
}"""), null);
    }

}

File: javascript/json-transform/src/__tests__/functions/TransformerFunctionJwtParse.test.ts
import { describe, test } from "vitest";
import { assertTransformation } from "../BaseTransformationTest";

describe("TransformerFunctionJwtParse", () => {
  test("inline", async () => {
    const testJWT =
      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c";
    await assertTransformation(testJWT, "$$jwtparse:$", {
      sub: "1234567890",
      name: "John Doe",
      iat: 1516239022,
    });
    //invalid
    await assertTransformation(true, "$$jwtparse:$", null);
  });

  test("object", async () => {
    const testJWT =
      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c";
    await assertTransformation(
      testJWT,
      {
        $$jwtparse: "$",
      },
      {
        sub: "1234567890",
        name: "John Doe",
        iat: 1516239022,
      },
    );
    //invalid
    await assertTransformation(
      true,
      {
        $$jwtparse: "$",
      },
      null,
    );
  });
});
